src/ir/effects.h

```diff
@@ -432,8 +432,18 @@ class EffectAnalyzer {
         auto iter = parent.funcEffectsMap->find(curr->target);
         if (iter != parent.funcEffectsMap->end()) {
           // We have effect information for this call target, and can just use
-          // that.
-          parent.mergeIn(iter->second);
+          // that. The one change we may want to make is to remove throws_, if
+          // the target function throws and we know that will be caught anyhow,
+          // the same as the code below for the general path.
+          const auto& targetEffects = iter->second;
+          if (targetEffects.throws_ && parent.tryDepth > 0) {
+            auto filteredEffects = targetEffects;
+            filteredEffects.throws_ = false;
+            parent.mergeIn(filteredEffects);
+          } else {
+            // Just merge in all the effects.
+            parent.mergeIn(targetEffects);
+          }
```

Comment:
Not sure if this is necessary. We don't set `throws_` if `tryDepth > 0` in the first place. https://github.com/WebAssembly/binaryen/blob/2fdb22bc26185e94ccd775bbfe8ea271be03df45/src/ir/effects.h#L368-L371

Reply:
That code just handles delegates, I think?

In the testcase I added there isn't a delegate, but the problem still happens. Basically the `visitCall` logic should not set `throws_` if we are in a try-catch-all. We have logic for that in the old code here:

https://github.com/WebAssembly/binaryen/blob/2fdb22bc26185e94ccd775bbfe8ea271be03df45/src/ir/effects.h#L426-L429

The new code above just does the same for when we use global effects.

Reply:
Ah right, I was confused. I thought the `tryDepth` was the target's...